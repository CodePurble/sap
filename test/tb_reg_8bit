#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x561e583d5b40 .scope module, "tb_reg_8bit" "tb_reg_8bit" 2 1;
 .timescale 0 0;
v0x561e583ff760_0 .var "async_reset", 0 0;
v0x561e583ff820_0 .var "clk", 0 0;
v0x561e583ff8f0_0 .var "i_en", 0 0;
v0x561e583ff9f0_0 .var "in", 7 0;
v0x561e583ffac0_0 .var "o_en", 0 0;
v0x561e583ffc00_0 .net "out", 7 0, L_0x561e583ffcf0;  1 drivers
S_0x561e583d5cd0 .scope module, "uut" "reg_8bit" 2 6, 3 2 0, S_0x561e583d5b40;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "i_en";
    .port_info 3 /INPUT 1 "o_en";
    .port_info 4 /INPUT 1 "async_reset";
    .port_info 5 /INPUT 1 "clk";
v0x561e583ff190_0 .net "async_reset", 0 0, v0x561e583ff760_0;  1 drivers
v0x561e583ff250_0 .net "clk", 0 0, v0x561e583ff820_0;  1 drivers
v0x561e583ff310_0 .var "hold", 7 0;
v0x561e583ff3b0_0 .net "i_en", 0 0, v0x561e583ff8f0_0;  1 drivers
v0x561e583ff450_0 .net "in", 7 0, v0x561e583ff9f0_0;  1 drivers
v0x561e583ff580_0 .net "o_en", 0 0, v0x561e583ffac0_0;  1 drivers
v0x561e583ff620_0 .net "out", 7 0, L_0x561e583ffcf0;  alias, 1 drivers
E_0x561e583d7200 .event posedge, v0x561e583ff190_0, v0x561e583ff250_0;
S_0x561e583d5e60 .scope module, "buffer" "tribuf_8bit" 3 11, 4 1 0, S_0x561e583d5cd0;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "enable";
o0x7ffbee86c018 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
; Elide local net with no drivers, v0x561e583d6680_0 name=_ivl_0
v0x561e583feeb0_0 .net "enable", 0 0, v0x561e583ffac0_0;  alias, 1 drivers
v0x561e583fef70_0 .net "in", 7 0, v0x561e583ff310_0;  1 drivers
v0x561e583ff030_0 .net "out", 7 0, L_0x561e583ffcf0;  alias, 1 drivers
L_0x561e583ffcf0 .functor MUXZ 8, v0x561e583ff310_0, o0x7ffbee86c018, v0x561e583ffac0_0, C4<>;
    .scope S_0x561e583d5cd0;
T_0 ;
    %wait E_0x561e583d7200;
    %load/vec4 v0x561e583ff3b0_0;
    %cmpi/e 0, 0, 1;
    %jmp/0xz  T_0.0, 4;
    %load/vec4 v0x561e583ff450_0;
    %assign/vec4 v0x561e583ff310_0, 0;
T_0.0 ;
    %load/vec4 v0x561e583ff190_0;
    %cmpi/e 1, 0, 1;
    %jmp/0xz  T_0.2, 4;
    %pushi/vec4 0, 0, 8;
    %assign/vec4 v0x561e583ff310_0, 0;
T_0.2 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0x561e583d5b40;
T_1 ;
    %vpi_call 2 10 "$dumpfile", "../simulation/tb_reg_8bit.vcd" {0 0 0};
    %vpi_call 2 11 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x561e583d5b40 {0 0 0};
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x561e583ff760_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ff8f0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x561e583ffac0_0, 0, 1;
    %pushi/vec4 255, 0, 8;
    %store/vec4 v0x561e583ff9f0_0, 0, 8;
    %delay 10, 0;
    %pushi/vec4 170, 0, 8;
    %store/vec4 v0x561e583ff9f0_0, 0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ffac0_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ff760_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ff8f0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ffac0_0, 0, 1;
    %delay 10, 0;
    %end;
    .thread T_1;
    .scope S_0x561e583d5b40;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x561e583ff820_0, 0, 1;
    %pushi/vec4 5, 0, 32;
T_2.0 %dup/vec4;
    %pushi/vec4 0, 0, 32;
    %cmp/s;
    %jmp/1xz T_2.1, 5;
    %jmp/1 T_2.1, 4;
    %pushi/vec4 1, 0, 32;
    %sub;
    %load/vec4 v0x561e583ff820_0;
    %inv;
    %store/vec4 v0x561e583ff820_0, 0, 1;
    %delay 10, 0;
    %jmp T_2.0;
T_2.1 ;
    %pop/vec4 1;
    %end;
    .thread T_2;
    .scope S_0x561e583d5b40;
T_3 ;
    %vpi_call 2 41 "$monitor", "clk = %b | in = %b | out = %b", v0x561e583ff820_0, v0x561e583ff9f0_0, v0x561e583ffc00_0 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "/home/ramprakash/repos/sap/test/tb_reg_8bit.v";
    "../hdl/reg_8bit.v";
    "../hdl/tribuf_8bit.v";
