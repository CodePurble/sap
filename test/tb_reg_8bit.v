module tb_reg_8bit;
	reg [7:0] in;
	reg low_i_en, low_o_en, async_reset, clk;
	wire [7:0] out;

	reg_8bit uut(.in(in), .out(out), .low_i_en(low_i_en), .low_o_en(low_o_en), .async_reset(async_reset), .clk(clk));

	initial
	begin
		$dumpfile("../simulation/tb_reg_8bit.vcd");
		$dumpvars(0, tb_reg_8bit);

		async_reset = 1'b1;
		low_i_en = 1'b0;
		low_o_en = 1'b1;
		in = 8'b11111111;
		#10;

		in = 8'b10101010;
		low_o_en = 1'b0;
		#10;

		async_reset = 1'b0;
		low_i_en = 1'b0;
		low_o_en = 1'b0;
		#10;
	end

	initial
	begin
		clk = 1'b0;
		repeat(5)
		begin
			clk = ~clk;
			#10;
		end
	end

	initial
	begin
		$monitor("low_o_en = %b | in = %b | out = %b", low_o_en, in, out);
	end
endmodule
