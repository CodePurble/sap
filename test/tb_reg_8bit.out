#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x555a838b7b40 .scope module, "tb_reg_8bit" "tb_reg_8bit" 2 1;
 .timescale 0 0;
v0x555a838e1730_0 .var "async_reset", 0 0;
v0x555a838e17f0_0 .var "clk", 0 0;
v0x555a838e18c0_0 .var "i_en", 0 0;
v0x555a838e19c0_0 .var "in", 7 0;
v0x555a838e1a90_0 .var "o_en", 0 0;
v0x555a838e1bd0_0 .net "out", 7 0, L_0x555a838e1cc0;  1 drivers
S_0x555a838b7cd0 .scope module, "uut" "reg_8bit" 2 6, 3 2 0, S_0x555a838b7b40;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "i_en";
    .port_info 3 /INPUT 1 "o_en";
    .port_info 4 /INPUT 1 "async_reset";
    .port_info 5 /INPUT 1 "clk";
v0x555a838e1160_0 .net "async_reset", 0 0, v0x555a838e1730_0;  1 drivers
v0x555a838e1220_0 .net "clk", 0 0, v0x555a838e17f0_0;  1 drivers
v0x555a838e12e0_0 .var "hold", 7 0;
v0x555a838e1380_0 .net "i_en", 0 0, v0x555a838e18c0_0;  1 drivers
v0x555a838e1420_0 .net "in", 7 0, v0x555a838e19c0_0;  1 drivers
v0x555a838e1550_0 .net "o_en", 0 0, v0x555a838e1a90_0;  1 drivers
v0x555a838e15f0_0 .net "out", 7 0, L_0x555a838e1cc0;  alias, 1 drivers
E_0x555a838b9200 .event posedge, v0x555a838e1160_0, v0x555a838e1220_0;
S_0x555a838b8680 .scope module, "buffer" "tribuf_8bit" 3 11, 4 2 0, S_0x555a838b7cd0;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "in";
    .port_info 1 /OUTPUT 8 "out";
    .port_info 2 /INPUT 1 "enable";
o0x7ff3d6742018 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
; Elide local net with no drivers, v0x555a838b88b0_0 name=_ivl_0
v0x555a838e0e80_0 .net "enable", 0 0, v0x555a838e1a90_0;  alias, 1 drivers
v0x555a838e0f40_0 .net "in", 7 0, v0x555a838e12e0_0;  1 drivers
v0x555a838e1000_0 .net "out", 7 0, L_0x555a838e1cc0;  alias, 1 drivers
L_0x555a838e1cc0 .functor MUXZ 8, v0x555a838e12e0_0, o0x7ff3d6742018, v0x555a838e1a90_0, C4<>;
    .scope S_0x555a838b7cd0;
T_0 ;
    %wait E_0x555a838b9200;
    %load/vec4 v0x555a838e1380_0;
    %cmpi/e 0, 0, 1;
    %jmp/0xz  T_0.0, 4;
    %load/vec4 v0x555a838e1420_0;
    %assign/vec4 v0x555a838e12e0_0, 0;
T_0.0 ;
    %load/vec4 v0x555a838e1160_0;
    %cmpi/e 1, 0, 1;
    %jmp/0xz  T_0.2, 4;
    %pushi/vec4 0, 0, 8;
    %assign/vec4 v0x555a838e12e0_0, 0;
T_0.2 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0x555a838b7b40;
T_1 ;
    %vpi_call 2 10 "$dumpfile", "../simulation/tb_reg_8bit.vcd" {0 0 0};
    %vpi_call 2 11 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x555a838b7b40 {0 0 0};
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x555a838e1730_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e18c0_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x555a838e1a90_0, 0, 1;
    %pushi/vec4 255, 0, 8;
    %store/vec4 v0x555a838e19c0_0, 0, 8;
    %delay 10, 0;
    %pushi/vec4 170, 0, 8;
    %store/vec4 v0x555a838e19c0_0, 0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e1a90_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e1730_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e18c0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e1a90_0, 0, 1;
    %delay 10, 0;
    %end;
    .thread T_1;
    .scope S_0x555a838b7b40;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x555a838e17f0_0, 0, 1;
    %pushi/vec4 5, 0, 32;
T_2.0 %dup/vec4;
    %pushi/vec4 0, 0, 32;
    %cmp/s;
    %jmp/1xz T_2.1, 5;
    %jmp/1 T_2.1, 4;
    %pushi/vec4 1, 0, 32;
    %sub;
    %load/vec4 v0x555a838e17f0_0;
    %inv;
    %store/vec4 v0x555a838e17f0_0, 0, 1;
    %delay 10, 0;
    %jmp T_2.0;
T_2.1 ;
    %pop/vec4 1;
    %end;
    .thread T_2;
    .scope S_0x555a838b7b40;
T_3 ;
    %vpi_call 2 41 "$monitor", "clk = %b | in = %b | out = %b", v0x555a838e17f0_0, v0x555a838e19c0_0, v0x555a838e1bd0_0 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "tb_reg_8bit.v";
    "../hdl/reg_8bit.v";
    "../hdl/tribuf_8bit.v";
